src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const ADD_OPTION = 'addProperties';
+  const REMOVE_OPTION = 'removeProperties';
+  const CLEAR_OPTION = 'clear';
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD_OPTION:
+        for (const key of Object.keys(action.extraData)) {
+          state[key] = action.extraData[key];
+        }
```

Comment:
Check out [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign). You can use it here to avoid a nested loop.

Reply:
[Checklist #5](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md)